File: algorithms/T/Truncate Sentence/Truncate Sentence.java
class Solution {
    public String truncateSentence(String s, int k) {
        int n = s.length();
        int count = 0;
        int i = 0;
        while(i<n){
            if(s.charAt(i)==' '){
                count++;
                if(count==k) 
                    return s.substring(0,i);
            }
            i++;
        }
        return s;
    }
}

File: scripts/algorithms/T/Truncate Sentence/Truncate Sentence.cpp
class Solution {
public:
	string truncateSentence(string s, int k) {

		int n = s.size();

		string ans;
		int cnt = 0;
		for(int i=0 ; i<n ; i++){

			if(s[i] == ' '){
				cnt++;
			}

			if(k == cnt){
				return ans;
			}

			ans += s[i];

		}

		return ans; 
	}
};